### PR TITLE
EM-2395: Alter email notification field to not use the primitive bool…

### PR DESCRIPTION
**Pull request by tomxelliott**
…ean type

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/em/orchestrator/automatedbundling/configuration/BundleConfiguration.java [hunk 1]
```diff
@@ -52,6 +52,6 @@ public BundleConfiguration(@JsonProperty("title") String title,
         this.documents = documents == null ? new ArrayList<>() : documents;
         this.paginationStyle = paginationStyle;
         this.documentNameValue = documentNameValue;
-        this.enableEmailNotification = enableEmailNotification;
+        this.enableEmailNotification = enableEmailNotification != null ? enableEmailNotification : null;
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/reform/em/orchestrator/automatedbundling/configuration/BundleConfiguration.java]: that's weird.... why is that necessary?

File: src/main/java/uk/gov/hmcts/reform/em/orchestrator/service/dto/CcdBundleDTO.java [hunk 2]
```diff
@@ -228,12 +231,16 @@ public void setEnableEmailNotification(CcdBoolean enableEmailNotification) {
...
+        } else {
+            this.enableEmailNotification = enableEmailNotification ? CcdBoolean.Yes : CcdBoolean.No;
+        }
     }
 
     @JsonIgnore
-    public boolean getEnableEmailNotificationAsBoolean() {
-        return enableEmailNotification == CcdBoolean.Yes ? true : false;
+    public Boolean getEnableEmailNotificationAsBoolean() {
+        return enableEmailNotification != null ? enableEmailNotification == CcdBoolean.Yes : null;
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/reform/em/orchestrator/service/dto/CcdBundleDTO.java]: I am wondering what happens if the field is present in the CCD definition file but set to false. With this code it will be null and omitted when sent back to CCD.... is that going to be okay? Or perhaps, if it is set to false we should return false.

File: src/main/java/uk/gov/hmcts/reform/em/orchestrator/automatedbundling/configuration/BundleConfiguration.java (hunk 1 above)
Comment (tomxelliott) [reply src/main/java/uk/gov/hmcts/reform/em/orchestrator/automatedbundling/configuration/BundleConfiguration.java]: Sorry, looked back at this now and you're right. It is unnecessary.
